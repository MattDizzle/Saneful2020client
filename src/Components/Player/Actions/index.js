import getFoodFromFridge from './getFoodFromFridge';
import relaxOnCouch from './relaxOnCouch';
import workOnComp from './workOnComp';


const DetermineAction = (actionString, executeAction) => {
  if (actionString === 'do work on the computer') {
    executeAction(workOnComp());
  }
  else if (actionString === 'eat food') {
    executeAction(getFoodFromFridge());
  }
  else if (actionString === 'relax on the couch') {
    executeAction(relaxOnCouch());
  }
};

export default DetermineAction;