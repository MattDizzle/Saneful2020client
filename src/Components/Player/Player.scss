:root{
  --top-int: 2;
  --left-int: 0;
}

.Player{
  width: 10vw;
  max-width: 52px;
  transition: 0.5s;
  z-index: 10;
  position: absolute;
  top: calc(-3vw + calc(10vw * var(--top-int)));
  left: calc(1vw + calc(10vw * var(--left-int)));
}


@media only screen and (min-width: 520px) {
  .Player{
    top: calc(-11px + calc(52px * var(--top-int)));
    left: calc(3px + calc(52px * var(--left-int)));
  }
}