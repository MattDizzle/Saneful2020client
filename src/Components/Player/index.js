import React from 'react';

import './Player.scss';

const Player = (props) => {

  return (
    <div className='player-animation'>
      <img className='Player' src={props.currentFrame} alt='player'></img>
    </div>
  );
};

export default Player;