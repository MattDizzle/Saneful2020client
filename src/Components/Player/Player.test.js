import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Player from './index';


describe(`Player Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <Player />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <Player />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})