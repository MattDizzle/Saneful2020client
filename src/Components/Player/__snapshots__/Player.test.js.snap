// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="player-animation"
>
  <img
    alt="player"
    className="Player"
  />
</div>
`;
